### PR TITLE
Sets up initial test module structure, container test rule and initial Skaffold file tests.

### PR DESCRIPTION
**Pull request by ivanporty**
Works towards #37, and #19.

Initializes unit tests rule for using IDE project and module basic test fixture, similar to [CloudToolsRule](https://github.com/GoogleCloudPlatform/google-cloud-intellij/blob/master/common-test-lib/src/main/java/com/google/cloud/tools/intellij/testing/CloudToolsRule.java).

For the start, test fixtures are always "light", i.e. giving minimal IDE infrastructure with slimmed down project and module support which should speed up the tests and be enough for majority of the unit tests.

Unit test names follow the convention most Kotlin tests follow.

Next PR will also add unit tests for project-based YAML search once I figure out proper way to add files to the mock project.

### CODE REVIEW

File: skaffold/src/test/kotlin/com/google/container/tools/skaffold/SkaffoldFilesTest.kt [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    @Test
+    fun `valid skaffold file is accepted`() {
+        val skaffoldFile = MockVirtualFile.file("skaffold.yaml")
+        skaffoldFile.setText("apiVersion: skaffold/v1alpha2")
+
+        assertThat(isSkaffoldFile(skaffoldFile)).isTrue()
+    }
+
+    @Test
+    fun `frivolously formatted skaffold file is accepted`() {
```
Comment (etanshaul) [inline skaffold/src/test/kotlin/com/google/container/tools/skaffold/SkaffoldFilesTest.kt]: maybe be more specific than "frivolously"? like something about whitespace if thats what you are testing.

File: common-test-lib/src/main/kotlin/com/google/container/tools/test/ContainerToolsRule.kt [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import org.junit.rules.TestRule
+import org.junit.runner.Description
+import org.junit.runners.model.Statement
+
+/**
+ * A custom [TestRule] for Container Tools unit tests.
+ *
+ * This rule adds the following functionality:
+ *
+ *  * Creates an [IdeaProjectTestFixture] and makes it available for tests via property. By default
```
Comment (etanshaul) [inline common-test-lib/src/main/kotlin/com/google/container/tools/test/ContainerToolsRule.kt]: doc formatting seems off.

Also do kotlin docs need a `<p>` tag like java docs after a new paragraph?

File: common-test-lib/src/main/kotlin/com/google/container/tools/test/ContainerToolsRule.kt [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+ *    this text fixture is "light", i.e. instance of [LightIdeaTestFixture].
+ */
+class ContainerToolsRule : TestRule {
+    lateinit var ideaProjectTestFixture: IdeaProjectTestFixture
+
+    override fun apply(baseStatement: Statement, description: Description): Statement {
+        return object : Statement() {
```
Comment (etanshaul) [inline common-test-lib/src/main/kotlin/com/google/container/tools/test/ContainerToolsRule.kt]: probably just my lack of understanding, but is the `object : Statement()` necessary? could you not just return `Statement()`?

File: common-test-lib/src/main/kotlin/com/google/container/tools/test/ContainerToolsRule.kt [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+ *
+ * This rule adds the following functionality:
+ *
+ *  * Creates an [IdeaProjectTestFixture] and makes it available for tests via property. By default
+ *    this text fixture is "light", i.e. instance of [LightIdeaTestFixture].
+ */
+class ContainerToolsRule : TestRule {
+    lateinit var ideaProjectTestFixture: IdeaProjectTestFixture
+
+    override fun apply(baseStatement: Statement, description: Description): Statement {
```
Comment (etanshaul) [inline common-test-lib/src/main/kotlin/com/google/container/tools/test/ContainerToolsRule.kt]: this function returns a single expression so you can probably use the `fun apply(..): Statement =` notation

File: common-test-lib/src/main/kotlin/com/google/container/tools/test/ContainerToolsRule.kt (hunk 2 above)
Comment (ivanporty) [reply common-test-lib/src/main/kotlin/com/google/container/tools/test/ContainerToolsRule.kt]: I believe the Kdoc is markdown (see https://kotlinlang.org/docs/reference/kotlin-doc.html, section on formatting), so this generates an item list. Seems to be rendered correctly by IDEA when you look it up as a quick doc. I don't think Kdoc supports html tags.

File: skaffold/src/test/kotlin/com/google/container/tools/skaffold/SkaffoldFilesTest.kt (hunk 1 above)
Comment (ivanporty) [reply skaffold/src/test/kotlin/com/google/container/tools/skaffold/SkaffoldFilesTest.kt]: sounds good, changed to a more specific name.

File: common-test-lib/src/main/kotlin/com/google/container/tools/test/ContainerToolsRule.kt (hunk 2 above)
Comment (etanshaul) [reply common-test-lib/src/main/kotlin/com/google/container/tools/test/ContainerToolsRule.kt]: ah so the double `*` is intentional then. sg, if so.

File: skaffold/src/test/kotlin/com/google/container/tools/skaffold/SkaffoldFilesTest.kt [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+import com.google.common.truth.Truth.assertThat
+import com.google.container.tools.test.ContainerToolsRule
+import com.intellij.mock.MockVirtualFile
+import org.junit.Rule
+import org.junit.Test
+
+/** Unit tests for Skaffold files functionality in [SkaffoldFiles.kt] */
+class SkaffoldFilesTest {
+    @get:Rule
```
Comment (etanshaul) [inline skaffold/src/test/kotlin/com/google/container/tools/skaffold/SkaffoldFilesTest.kt]: just for my ignorance. what is this `@get` syntax? thanks

File: skaffold/src/test/kotlin/com/google/container/tools/skaffold/SkaffoldFilesTest.kt [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    val containerToolsRule = ContainerToolsRule()
```
Comment (etanshaul) [inline skaffold/src/test/kotlin/com/google/container/tools/skaffold/SkaffoldFilesTest.kt]: is this rule even needed for this particular unit test class?

File: common-test-lib/src/main/kotlin/com/google/container/tools/test/ContainerToolsRule.kt (hunk 3 above)
Comment (ivanporty) [reply common-test-lib/src/main/kotlin/com/google/container/tools/test/ContainerToolsRule.kt]: This is the syntax required to return an instance of a newly created anonymous class, object meaning we return an instance and then class structure. I looked up and didn't find more elegant ways of doing it. Posted request for review into Kotlin readability group, hopefully someone will review from their side.

**Comment by ivanporty** [comment]
Posted new request for review on Kotlin readability group.

File: skaffold/src/test/kotlin/com/google/container/tools/skaffold/SkaffoldFilesTest.kt [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    @Test
+    fun `valid skaffold file is accepted`() {
+        val skaffoldFile = MockVirtualFile.file("skaffold.yaml")
+        skaffoldFile.setText("apiVersion: skaffold/v1alpha2")
+
+        assertThat(isSkaffoldFile(skaffoldFile)).isTrue()
+    }
+
+    @Test
+    fun `extra spaces and tabs in header skaffold file is accepted`() {
```
[the remark this answers is not in the document]
Comment (ivanporty) [reply skaffold/src/test/kotlin/com/google/container/tools/skaffold/SkaffoldFilesTest.kt]: done.

**Comment by ivanporty** [comment]
@keyboardsurfer kindly agreed to help with Kotlin review, please take a look at this PR.

File: skaffold/src/main/kotlin/com/google/container/tools/skaffold/SkaffoldFiles.kt [hunk 8]
```diff
@@ -35,7 +35,7 @@ private val SKAFFOLD_API_HEADER_PATTERN: Pattern by lazy {
 /**
  * Checks if a given file is a valid Skaffold configuration file based on type and API version.
  */
-internal fun isSkaffoldFile(file: VirtualFile): Boolean {
+fun isSkaffoldFile(file: VirtualFile): Boolean {
```
Comment (ivanporty) [inline skaffold/src/main/kotlin/com/google/container/tools/skaffold/SkaffoldFiles.kt]: Question for Kotlin reviewer - we cannot make this work as `internal` with test code (default test code by Gradle conventions), although it should work according to documentation. Can you please hint us where the issue could be?

Comment (keyboardsurfer) [reply skaffold/src/main/kotlin/com/google/container/tools/skaffold/SkaffoldFiles.kt]: I haven't seen this causing issues just yet. Agreed that, according to docs this should work.

Comment (ivanporty) [reply skaffold/src/main/kotlin/com/google/container/tools/skaffold/SkaffoldFiles.kt]: I guess we will need to investigate this further, I suspect this has something to do with Gradle sub-modules. Thanks!

Comment (ivanporty) [reply skaffold/src/main/kotlin/com/google/container/tools/skaffold/SkaffoldFiles.kt]: Filed #43

**Pull request merged**